### DISCUSSION TITLE
Radar Mosaic: How Py-ART interpolate two radars?

### DISCUSSION THREAD

**Discussion by MarceloTSilveira** [ARM-DOE/pyart#1365] [Q&A] [answered]
I would like to know how Py-ART interpolate two radar data when generating a grid (grid_from_radars function). To be more specific, if Py-ART gives the same weight to the radars on each grid point or gives more weight to the closest radar.

For example, I'm creating a mosaic grid from two radars data located at Rio de Janeiro, Brazil. It seems the resulting grid is giving more weight for each point to the closest radar. This can be noticed on the following images, where the place signed by a gray arrow is the location of the first radar and a hole appears right over it, although the second radar contains data over that place, which is being ignored. 

First image: grid images at 2000 m from each radar generated separatelly.
![Radars_Separated](https://user-images.githubusercontent.com/114624627/209567373-0f5773fa-5513-48b1-89b9-29cf54e0cc8f.gif)

Second image: grid image at 2000 m from radar mosaic.
![Mosaic](https://user-images.githubusercontent.com/114624627/209567808-8e087d91-df1e-4bfc-afc1-1ef5b22b0a93.gif)

Notice that the mosaic image keeps the hole (see under the gray arrow) instead of filling it with the information from the second radar (image 1, right).

Thanks for helping.

**Comment by mgrover1** [comment]
Could you please provide an example of the code you are using to grid the radars, along with a couple of sample files? This will help diagnose the issue here.

**Reply by MarceloTSilveira** [reply]
Code for gerenating the image mosaic:

```
import pyart
import numpy as np
import warnings

warnings.filterwarnings('ignore')


#
# Configuration
#

grid_shape = (1, 571, 681)
grid_limits = ((2000, 2000), (-250000, 320000), (-250000, 430000))
radar_fields = ['DBZH', 'rain_rate']


#
# Radar pair file list
#

pair_list = [['9921GUA-20220516-204010-PPIVol-0000.nc', '9915MAC-20220516-204000-PPIVol-0000.nc'],
             ['9921GUA-20220516-204510-PPIVol-0000.nc', '9915MAC-20220516-204500-PPIVol-0000.nc']]
out_file_list = ['Pair_20220516_2040.nc', 'Pair_20220516_2045.nc']


#
# Process radar data
#

def process_radar(pos):
    # Lê imagem do radar
    print(' Reading radar raw data ...')
    radar_list = [pyart.io.read(file) for file in pair_list[pos]]
    
    gf = [0, 0]
    
    for i, radar in enumerate(radar_list):
        # Create filters
        gf[i] = pyart.correct.GateFilter(radar)
        gf[i].exclude_below('RHOHV', 0.80)
        gf[i].exclude_below('DBZH', 0)
        
        # Estimate precipitation rate
        a, b = 200, 1.6
        print(' Estimating precipitation rate ...')
        rain_rate = (10.0 ** (radar.fields['DBZH']['data'] / 10.0) / a) ** (1.0/b)
        rr_dict = {'units' : 'mm/h', 'standard_name' : 'rain_rate_mp', 'long_name' : 'Rain Rate - Marshall-Palmer', 
                   'coordinates': 'elevation azimuth range', 'data': rain_rate}
        radar.add_field('rain_rate', rr_dict, True)
    
    # Create 2000 m CAPPI grid
    print(' Creating CAPPI ...')
    grid = pyart.map.grid_from_radars(radar_list, grid_shape=grid_shape, grid_limits=grid_limits, 
                                      fields=radar_fields, gridding_algo="map_gates_to_grid", 
                                      weighting_function='BARNES2', gatefilters=gf)
    
    # Save file
    print(' Saving file ...')
    grid.write(out_file_list[pos])


for i, file in enumerate(pair_list):
    print('Processessing radar image {} / {} ...'.format(i+1, len(pair_list)))
    process_radar(i)
```

The sample files can be found here:
https://github.com/MarceloTSilveira/radar

Thank you for helping us.

**Reply by mgrover1** [reply]
What type of format are those .rar files?

**Reply by kmuehlbauer** [reply]
@mgrover1 Most likely packed files with the [rar-algorithm](https://en.m.wikipedia.org/wiki/RAR_(file_format)).

**Reply by mgrover1** [reply]
Okay! @MarceloTSilveira can you share the nc files you read in the script? Unfortunately I cannot read the rar files.

**Reply by MarceloTSilveira** [reply]
I've changed the files to ZIP format. Please, check them out again.

**Comment by mgrover1** [comment, marked as answer]
Taking a look at this @MarceloTSilveira , I think the issue is with the missing values + radius of influence.

Here are the results from gridding from a single radar (the one that is farther away, that does not have missing values from the convective cell you highlighted)

![Screenshot 2023-01-05 at 9 44 07 AM](https://user-images.githubusercontent.com/26660300/210822759-ec26bf1c-3ee7-4e64-bb79-a48b3dda279d.png)

One thing to note here is that the ROI is quite high, ~3 to 4 km

When you add in the other radar, we see the ROI decreases, and it pulls from the closer radar that has missing values, and the valid values are too far away to be considered "valid" using the Barnes scheme

![Screenshot 2023-01-05 at 9 42 05 AM](https://user-images.githubusercontent.com/26660300/210822898-2bf67d2d-3abc-4711-97be-90ac9746b056.png)

One way around this is to set your own radius of influence, so you can pull from the radar you have data for in that region, as shown below with the gridding parameters

```python
grid = pyart.map.grid_from_radars(radar_list, grid_shape=grid_shape, grid_limits=grid_limits, 
                                      fields=radar_fields, gridding_algo="map_gates_to_grid", constant_roi=3000, gatefilters=gf)
```
![Screenshot 2023-01-05 at 9 56 25 AM](https://user-images.githubusercontent.com/26660300/210824132-12dff912-274d-4e53-8d89-213db76bda40.png)

One caveat with this approach is that there is more smoothing that might be desired, but it does deal with that gap in the data (a constant pro/con when looking at different gridding approaches).

I hope this helps!

**Reply by MarceloTSilveira** [reply]
It looks very nice now.
Thank you so much for helping.
